config/software/openssl.rb

```diff
@@ -17,135 +16,137 @@
+version("1.0.1t") { source md5: "9837746fcf8a6727d46d22ca35953da1" }
+version("1.0.1s") { source md5: "562986f6937aabc7c11a6d376d8a0d26" }
+version("1.0.1r") { source md5: "1abd905e079542ccae948af37e393d28" }
 
 relative_path "openssl-#{version}"
 
 build do
-  patch :source => "openssl-1.0.2i-do-not-build-docs.patch"
-
-  env = case ohai["platform"]
-        when "mac_os_x"
-          {
-            "CFLAGS" => "-arch x86_64 -m64 -L#{install_dir}/embedded/lib -I#{install_dir}/embedded/include -I#{install_dir}/embedded/include/ncurses",
-            "LDFLAGS" => "-arch x86_64 -R#{install_dir}/embedded/lib -L#{install_dir}/embedded/lib -I#{install_dir}/embedded/include -I#{install_dir}/embedded/include/ncurses",
-          }
-        when "aix"
-          {
-            "CC" => "xlc -q64",
-            "CXX" => "xlC -q64",
-            "LD" => "ld -b64",
-            "CFLAGS" => "-q64 -I#{install_dir}/embedded/include -O",
-            "CXXFLAGS" => "-q64 -I#{install_dir}/embedded/include -O",
-            "LDFLAGS" => "-q64 -L#{install_dir}/embedded/lib -Wl,-blibpath:#{install_dir}/embedded/lib:/usr/lib:/lib",
-            "OBJECT_MODE" => "64",
-            "AR" => "/usr/bin/ar",
-            "ARFLAGS" => "-X64 cru",
-            "M4" => "/opt/freeware/bin/m4",
-          }
-        when "solaris2"
-          {
-            "CFLAGS" => "-L#{install_dir}/embedded/lib -I#{install_dir}/embedded/include",
-            "LDFLAGS" => "-R#{install_dir}/embedded/lib -L#{install_dir}/embedded/lib -I#{install_dir}/embedded/include -static-libgcc",
-            "LD_OPTIONS" => "-R#{install_dir}/embedded/lib",
-          }
-        else
-          {
-            "CFLAGS" => "-I#{install_dir}/embedded/include",
-            "LDFLAGS" => "-Wl,-rpath,#{install_dir}/embedded/lib -L#{install_dir}/embedded/lib",
-          }
-        end
 
-  common_args = [
+  env = with_standard_compiler_flags(with_embedded_path)
+  if aix?
+    env["M4"] = "/opt/freeware/bin/m4"
+  elsif freebsd?
+    # Should this just be in standard_compiler_flags?
+    env["LDFLAGS"] += " -Wl,-rpath,#{install_dir}/embedded/lib"
+  elsif windows?
+    # XXX: OpenSSL explicitly sets -march=i486 and expects that to be honored.
+    # It has OPENSSL_IA32_SSE2 controlling whether it emits optimized SSE2 code
+    # and the 32-bit calling convention involving XMM registers is...  vague.
+    # Do not enable SSE2 generally because the hand optimized assembly will
+    # overwrite registers that mingw expects to get preserved.
+    env["CFLAGS"] = "-I#{install_dir}/embedded/include"
+    env["CPPFLAGS"] = env["CFLAGS"]
+    env["CXXFLAGS"] = env["CFLAGS"]
+  end
+
+  configure_args = [
     "--prefix=#{install_dir}/embedded",
     "--with-zlib-lib=#{install_dir}/embedded/lib",
     "--with-zlib-include=#{install_dir}/embedded/include",
     "no-idea",
     "no-mdc2",
     "no-rc5",
-    "zlib",
     "shared",
-    "no-ssl3",
```

Comment:
we should probably keep this flag (see https://github.com/DataDog/omnibus-software/commit/75e8f2644f73fdef5124356ded6c7189fccf7e18)